eth/backend.go

```diff
@@ -126,6 +129,12 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 		}
 		config.TrieDirtyCache = 0
 	}
+	// Optimize memory distribution by reallocating surplus allowance from the
+	// dirty cache to the clean cache.
+	if config.StateScheme == rawdb.PathScheme && config.TrieDirtyCache > pathdb.MaxBufferSize/1024/1024 {
+		config.TrieCleanCache += config.TrieDirtyCache - pathdb.MaxBufferSize/1024/1024
+		config.TrieDirtyCache = pathdb.MaxBufferSize / 1024 / 1024
```

Comment:
Interestingly, this will prevent the warning message displaying that the buffer was capped. Should we display something here, or is the log.Info enough?